ci/goreleaser/goreleaser.yml

```diff
@@ -168,6 +170,54 @@ dockers:
       - go.sum
       - main.go
       - ci/images/plugin-compiler
+  # Build plugin-compiler container arm64
+  - ids:
+      - std
+    image_templates:
+      - "tykio/tyk-plugin-compiler:{{ .Tag }}-arm64"
+      - "tykio/tyk-plugin-compiler:v{{ .Major }}.{{ .Minor }}{{.Prerelease}}-arm64"
+    build_flag_templates:
+      - "--platform=linux/arm64"
+      - "--label=org.opencontainers.image.created={{.Date}}"
+      - "--label=org.opencontainers.image.title=tyk-plugin-compiler"
+      - "--label=org.opencontainers.image.revision={{.FullCommit}}"
+      - "--label=org.opencontainers.image.version={{.Version}}"
+      - "--build-arg=BASE_IMAGE=golang:{{ .Env.GO_VERSION }}"
```

Comment:
@titpetric Not very sure if this would work. IIRC, the reason we have a golang-cross image that's used across both the gateway build and plugin build is to maintain uniformity in the tool-chain. The go compiler, CC, build flags etc. should be uniform across both the original binary and the plugin build - otherwise the plugin might not get loaded.

Reply:
@asutosh it works in the sense that it produces a native arm64 build image, as golang-cross doesn't have a linux/arm64 platform manifest. There will be more testing in the area so we verify fidelity, but yeah, a native plugin build may be incompatible with the cross-compiled gateway, but it would be compatible with a native built gw.